frontend/src/components/Map/index.js

```diff
@@ -23,617 +32,640 @@ export default function Map() {
+      {!draggable && <div className={classNames(styles.fill, styles.mask)} />}
+    </div>
+  );
+}
+
+async function fetchCovidData(scope) {
+  return await api(`data/` + scope, {
+    method: "GET",
+  });
+}
 
-      // Last value is the default, used where there is no data
-      expression.push("rgba(0,0,0,0)");
+function addWorldLayer(map, data) {
+  data.then(function (covidData) {
+    const expression = covidData.map(function (row) {
+      return [row.country, get_color(row.group)];
+    });
 
-      const zoomThreshold = 4;
-      // Add layer from the vector tile source with data-driven style
-      map.addLayer(
-        {
-          id: "states-join",
-          type: "fill",
-          source: "states",
-          maxzoom: zoomThreshold,
-          "source-layer": "states",
-          paint: {
-            "fill-color": expression,
+    map.addLayer(
+      {
+        id: "maine",
+        type: "fill",
+        layout: {},
+        maxzoom: countryMinZoom,
+        paint: {
+          "fill-color": {
+            type: "categorical",
+            property: "name",
+            stops: expression,
           },
-        }
-        // 'road-label'
-      );
+          "fill-opacity": 0.7,
+          "fill-outline-color": fillOutlineColor,
+        },
+        source: {
+          type: "vector",
+          url: "mapbox://saurabhp.countries_tileset",
+        },
+        "source-layer": "countries",
+      },
+      "waterway-label"
+    );
+  });
+}
 
-      /*----------------------------  County Level Data ----------------------------------*/
+function get_color(group) {
+  return "rgba(" + group * 255 + ", " + 0 + ", " + 0 + ", 1)";
+}
 
-      // var countyData ={}
-      // var url ="http://13.57.220.143/getcounty";
-      // fetch(url,{ method: 'post'}).then((resp) => resp.json())
-      //   .then(function(data) {
-      //     countyData=data;
-      //     console.log(data);
-      //   })
-      //   .catch(function(error) {
-      //     console.log(error);
-      //   });
-      // TODO: I need help making this URL fetch work
+function addUSStatesLayer(map, data) {
+  const stateToFIPS = {
+    AK: "02",
+    AL: "01",
+    AR: "05",
+    AS: "60",
+    AZ: "04",
+    CA: "06",
+    CO: "08",
+    CT: "09",
+    DC: "11",
+    DE: "10",
+    FL: "12",
+    GA: "13",
+    GU: "66",
+    HI: "15",
+    IA: "19",
+    ID: "16",
+    IL: "17",
+    IN: "18",
+    KS: "20",
+    KY: "21",
+    LA: "22",
+    MA: "25",
+    MD: "24",
+    ME: "23",
+    MI: "26",
+    MN: "27",
+    MO: "29",
+    MS: "28",
+    MT: "30",
+    NC: "37",
+    ND: "38",
+    NE: "31",
+    NH: "33",
+    NJ: "34",
+    NM: "35",
+    NV: "32",
+    NY: "36",
+    OH: "39",
+    OK: "40",
+    OR: "41",
+    PA: "42",
+    PR: "72",
+    RI: "44",
+    SC: "45",
+    SD: "46",
+    TN: "47",
+    TX: "48",
+    UT: "49",
+    VA: "51",
+    VI: "78",
+    VT: "50",
+    WA: "53",
+    WI: "55",
+    WV: "54",
+    WY: "56",
```

Comment:
Again, not something to change now, but as a wild thought this might be less intrusive to sight if we organize it differently...

```javascript
[
    // ...an ordered array of strings for the states
].reduce((acc, state, index) => {
    acc[state] = `0${index + 1}`.slice(-2);
    return acc;
}, {})
```

Who knows how we'll be fetching this data in two months though

Reply:
I agree (for the future)